tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionReplaySettings.cs

```diff
@@ -20,15 +17,16 @@ internal class ExceptionReplaySettings
         public const int DefaultMaxFramesToCapture = 4;
         public const int DefaultRateLimitSeconds = 60 * 60; // 1 hour
         public const int DefaultMaxExceptionAnalysisLimit = 100;
+        private bool? _internalExceptionReplayEnabled;
 
         public ExceptionReplaySettings(IConfigurationSource? source, IConfigurationTelemetry telemetry)
         {
             source ??= NullConfigurationSource.Instance;
             var config = new ConfigurationBuilder(source, telemetry);
 
-#pragma warning disable CS0612 // Type or member is obsolete
-            Enabled = config.WithKeys(ConfigurationKeys.Debugger.ExceptionDebuggingEnabled, ConfigurationKeys.Debugger.ExceptionReplayEnabled).AsBool(false);
-#pragma warning restore CS0612 // Type or member is obsolete
+            Enabled = config.WithKeys(ConfigurationKeys.Debugger.ExceptionReplayEnabled).AsBool(false);
+
+            _internalExceptionReplayEnabled = config.WithKeys(ConfigurationKeys.Debugger.ExceptionReplayEnabled).AsBool();
```

Comment:
You are reading the same setting from _all_ of the sources, multiple times, this is expensive, and shouldn't be necessary. 

If you need to do this (I'm still not entirely convinced you do - you should always be able to enable via remote config) you can do it like this:

```suggestion
            var enabledResult = config.WithKeys(ConfigurationKeys.Debugger.ExceptionReplayEnabled).AsBool(false);

            Enabled = enabledResult.WithDefault(false);
            CanBeEnabled = !(enabledResult.ConfigurationResult.IsValid && enabledResult.ConfigurationResult.Result == false);
```

Note that I changed the name of the property (and removed the unecessary field), because it's currently confusing to have `ExceptionReplaySettings.Enabled` and `IsExceptionReplayDisabled` meaning different things